### ISSUE TITLE
okx: Parameter instId  error on fetch_positions

### ISSUE THREAD

**Issue by donmega** [freqtrade/freqtrade#7720]
## Describe your environment

  * Operating system: Ubuntu 22.04
  * Python Version: Docker developer freqai(`python -V`)
  * CCXT version: Docker developer freqai(`pip freeze | grep ccxt`)
  * Freqtrade Version: docker-2022.11.dev-8bc71f20 (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

I am having the following problem since a few days. I am running Freqai on the latest nightly. So since a few days i am receiving the following error with every sell action. I am also not receiving a Telegram notification (Only the exiting announcement, not the final exited message). Because of this my stoplossprotection does not trigger. The sell does happen tho on the exchange. I think this error began popping up since pull request 7701 from @robcaulk , but i am not 100% sure. Iprogger in the Freqtrade Discord also seems to have the same problem. I am using the OKEX exchange. It works good tho when running dry, this only happens in Live.

### Steps to reproduce:

  1. Use the latest docker nightly image of freqai
  2. Use the Okex exchange and wait for a sell action
  
### Observed Results:

  * What happened?
  * What did you expect to happen?

### Relevant code exceptions or logs

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```
  // paste your log here
freqtrade    | 2022-11-07 16:15:08,430 - uvicorn.error - ERROR - Exception in ASGI application
freqtrade    | Traceback (most recent call last):
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
freqtrade    |     result = await app(  # type: ignore[func-returns-value]
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
freqtrade    |     return await self.app(scope, receive, send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/uvicorn/middleware/message_logger.py", line 86, in __call__
freqtrade    |     raise exc from None
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/uvicorn/middleware/message_logger.py", line 82, in __call__
freqtrade    |     await self.app(scope, inner_receive, inner_send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/fastapi/applications.py", line 270, in __call__
freqtrade    |     await super().__call__(scope, receive, send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/applications.py", line 124, in __call__
freqtrade    |     await self.middleware_stack(scope, receive, send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
freqtrade    |     raise exc
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
freqtrade    |     await self.app(scope, receive, _send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 84, in __call__
freqtrade    |     await self.app(scope, receive, send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 75, in __call__
freqtrade    |     raise exc
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 64, in __call__
freqtrade    |     await self.app(scope, receive, sender)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
freqtrade    |     raise e
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
freqtrade    |     await self.app(scope, receive, send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/routing.py", line 680, in __call__
freqtrade    |     await route.handle(scope, receive, send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/routing.py", line 275, in handle
freqtrade    |     await self.app(scope, receive, send)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/routing.py", line 65, in app
freqtrade    |     response = await func(request)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/fastapi/routing.py", line 231, in app
freqtrade    |     raw_response = await run_endpoint_function(
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/fastapi/routing.py", line 162, in run_endpoint_function
freqtrade    |     return await run_in_threadpool(dependant.call, **values)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
freqtrade    |     return await anyio.to_thread.run_sync(func, *args)
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/anyio/to_thread.py", line 31, in run_sync
freqtrade    |     return await get_asynclib().run_sync_in_worker_thread(
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
freqtrade    |     return await future
freqtrade    |   File "/home/ftuser/.local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 867, in run
freqtrade    |     result = context.run(func, *args)
freqtrade    |   File "/freqtrade/freqtrade/rpc/api_server/api_v1.py", line 97, in status
freqtrade    |     return rpc._rpc_trade_status()
freqtrade    |   File "/freqtrade/freqtrade/rpc/rpc.py", line 221, in _rpc_trade_status
freqtrade    |     open_order='({} {} rem={:.8f})'.format(
freqtrade    | TypeError: unsupported format string passed to NoneType.__format__
freqtrade    | 2022-11-07 16:15:08,629 - freqtrade.exchange.common - WARNING - fetch_positions() returned exception: "Could not get positions due to BadRequest. Message: okx {"code":"51000","data":[],"msg":"Parameter instId  error"}". Retrying still for 4 times.
freqtrade    | 2022-11-07 16:15:08,830 - freqtrade.exchange.common - WARNING - fetch_positions() returned exception: "Could not get positions due to BadRequest. Message: okx {"code":"51000","data":[],"msg":"Parameter instId  error"}". Retrying still for 3 times.
freqtrade    | 2022-11-07 16:15:09,033 - freqtrade.exchange.common - WARNING - fetch_positions() returned exception: "Could not get positions due to BadRequest. Message: okx {"code":"51000","data":[],"msg":"Parameter instId  error"}". Retrying still for 2 times.
freqtrade    | 2022-11-07 16:15:09,244 - freqtrade.exchange.common - WARNING - fetch_positions() returned exception: "Could not get positions due to BadRequest. Message: okx {"code":"51000","data":[],"msg":"Parameter instId  error"}". Retrying still for 1 times.
freqtrade    | 2022-11-07 16:15:09,428 - freqtrade.exchange.common - WARNING - fetch_positions() returned exception: "Could not get positions due to BadRequest. Message: okx {"code":"51000","data":[],"msg":"Parameter instId  error"}". Giving up.
freqtrade    | 2022-11-07 16:15:09,428 - freqtrade.worker - WARNING - Error: Could not get positions due to BadRequest. Message: okx {"code":"51000","data":[],"msg":"Parameter instId  error"}, retrying in 30 seconds...

**Comment by xmatthias** [comment]
Are you able to share more logs from that bot?

the "final" log suggests a bug in ccxt - which was however fixed about a week ago in develop (#7667)  - and testing it right now again - is not in the ccxt version used by `8bc71f20`.

The same exact error will occur on startup (which also fetches open positions).
if it doesn't then the version that caused the error is not the one you're running now.

**Comment by donmega** [comment]
Hi @xmatthias 

Thanks for the fast reply. Here is some more information:

freqtrade    | 2022-11-07 17:48:47,140 - freqtrade - INFO - freqtrade docker-2022.11.dev-8bc71f20
freqtrade    | 2022-11-07 17:48:47,441 - freqtrade.worker - INFO - Starting worker docker-2022.11.dev-8bc71f20
freqtrade    | 2022-11-07 17:48:47,441 - freqtrade.configuration.load_config - INFO - Using config: /freqtrade/user_data/config.json ...
freqtrade    | 2022-11-07 17:48:47,443 - freqtrade.loggers - INFO - Verbosity set to 0
freqtrade    | 2022-11-07 17:48:47,443 - freqtrade.configuration.configuration - INFO - Runmode set to live.
freqtrade    | 2022-11-07 17:48:47,444 - freqtrade.configuration.configuration - INFO - Parameter --db-url detected ...
freqtrade    | 2022-11-07 17:48:47,444 - freqtrade.configuration.configuration - INFO - Dry run is disabled
freqtrade    | 2022-11-07 17:48:47,444 - freqtrade.configuration.configuration - INFO - Using DB: "sqlite:////freqtrade/user_data/tradesv3.sqlite"
freqtrade    | 2022-11-07 17:48:47,444 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 18 ...
freqtrade    | 2022-11-07 17:48:47,487 - freqtrade.configuration.configuration - INFO - Using user-data directory: /freqtrade/user_data ...
freqtrade    | 2022-11-07 17:48:47,488 - freqtrade.configuration.configuration - INFO - Using data directory: /freqtrade/user_data/data/okx ...
freqtrade    | 2022-11-07 17:48:47,488 - freqtrade.configuration.configuration - INFO - Using freqaimodel class name: XGBoostRegressor
freqtrade    | 2022-11-07 17:48:47,489 - freqtrade.exchange.check_exchange - INFO - Checking exchange...
freqtrade    | 2022-11-07 17:48:47,497 - freqtrade.exchange.check_exchange - INFO - Exchange "okx" is officially supported by the Freqtrade development team.
freqtrade    | 2022-11-07 17:48:47,497 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
freqtrade    | 2022-11-07 17:48:47,535 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy FreqaiExampleStrategy from '/freqtrade/user_data/strategies/FreqaiExampleStrategy.py'...
freqtrade    | 2022-11-07 17:48:47,535 - freqtrade.strategy.hyper - INFO - Found no parameter file.
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'order_types' with value in config file: {'entry': 'market', 'exit': 'market', 'force_exit': 'market', 'emergency_exit': 'market', 'trailing_stop_loss': 'market', 'stoploss': 'market', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}.
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: USDT.
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: unlimited.
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'unfilledtimeout' with value in config file: {'entry': 2, 'exit': 2, 'exit_timeout_count': 0, 'unit': 'minutes'}.
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'2400': -1}
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using timeframe: 15m
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.08
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: True
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive: 0.019
freqtrade    | 2022-11-07 17:48:47,536 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.02
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: True
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_custom_stoploss: False
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: True
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'entry': 'market', 'exit': 'market', 'force_exit': 'market', 'emergency_exit': 'market', 'trailing_stop_loss': 'market', 'stoploss': 'market', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'entry': 'GTC', 'exit': 'GTC'}
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: USDT
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: unlimited
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using protections: [{'method': 'StoplossGuard', 'lookback_period_candles': 5, 'trade_limit': 1, 'required_profit': -0.07, 'stop_duration': 120, 'only_per_pair': True}]
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 40
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using unfilledtimeout: {'entry': 2, 'exit': 2, 'exit_timeout_count': 0, 'unit': 'minutes'}
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_exit_signal: True
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_only: False
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_entry_signal: False
freqtrade    | 2022-11-07 17:48:47,537 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_offset: 0.0
freqtrade    | 2022-11-07 17:48:47,538 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using disable_dataframe_checks: False
freqtrade    | 2022-11-07 17:48:47,538 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_buying_expired_candle_after: 0
freqtrade    | 2022-11-07 17:48:47,538 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using position_adjustment_enable: False
freqtrade    | 2022-11-07 17:48:47,538 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using max_entry_position_adjustment: -1
freqtrade    | 2022-11-07 17:48:47,538 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
freqtrade    | 2022-11-07 17:48:47,543 - freqtrade.exchange.exchange - INFO - Using CCXT 2.1.33
freqtrade    | 2022-11-07 17:48:47,543 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'options': {'defaultType': 'swap'}}
freqtrade    | 2022-11-07 17:48:47,550 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'options': {'defaultType': 'swap'}}
freqtrade    | 2022-11-07 17:48:47,556 - freqtrade.exchange.exchange - INFO - Using Exchange "OKX"
freqtrade    | 2022-11-07 17:48:49,268 - freqtrade.exchange.exchange - INFO - Using cached leverage_tiers.
freqtrade    | 2022-11-07 17:48:49,268 - freqtrade.exchange.exchange - INFO - Done initializing 124 markets.
freqtrade    | 2022-11-07 17:48:49,372 - freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Okx'...

CCXT seems to be newest. Here is my docker-compose:

---
version: '3'
services:
  freqtrade:
    image: freqtradeorg/freqtrade:develop_freqai


Let me know if you need more information.

Thanks!

**Comment by xmatthias** [comment]
If that log continues - i assume it'll error before the first "Bot heartbeat" ?

**Comment by donmega** [comment]
> If that log continues - i assume it'll error before the first "Bot heartbeat" ?
[error.txt](https://github.com/freqtrade/freqtrade/files/9954563/error.txt)

It seems to happen only on sell actions as far as i can see. Please see the attachment for the full log

**Comment by donmega** [comment]
> If that log continues - i assume it'll error before the first "Bot heartbeat" ?
[error2.txt](https://github.com/freqtrade/freqtrade/files/9954643/error2.txt)

Here is another example. It happens in this example when selling TRX/USDT

**Comment by xmatthias** [comment]
you're right, the bug is still not fully fixed. opened https://github.com/ccxt/ccxt/issues/15589 for that.

**Comment by donmega** [comment]
> you're right, the bug is still not fully fixed. opened [ccxt/ccxt#15589](https://github.com/ccxt/ccxt/issues/15589) for that.

Thank you for passing it to the ccxt team. Looks like they've fixed it in version 2.1.49 of ccxt

**Comment by xmatthias** [comment]
Fixed in d3006f7f3e7f5e502bb8331f2b85bef089cff995
